python/ribasim/ribasim/validation.py

```diff
@@ -0,0 +1,83 @@
+# Table for connectivity
+# "Basin": ["LinearResistance"] means that the downstream of basin can be LinearResistance only
+connectivity: dict[str, list[str]] = {
+    "Basin": [
+        "LinearResistance",
+        "ManningResistance",
+        "TabulatedRatingCurve",
+        "Pump",
+        "Outlet",
+        "UserDemand",
+    ],
+    "LinearResistance": ["Basin", "LevelBoundary"],
+    "ManningResistance": ["Basin"],
+    "TabulatedRatingCurve": ["Basin", "Terminal", "LevelBoundary"],
+    "LevelBoundary": ["LinearResistance", "Pump", "Outlet", "TabulatedRatingCurve"],
+    "FlowBoundary": ["Basin", "Terminal", "LevelBoundary"],
+    "Pump": ["Basin", "Terminal", "LevelBoundary"],
+    "Outlet": ["Basin", "Terminal", "LevelBoundary"],
+    "Terminal": [],
+    "DiscreteControl": [
+        "Pump",
+        "Outlet",
+        "TabulatedRatingCurve",
+        "LinearResistance",
+        "ManningResistance",
+        "PidControl",
+    ],
+    "ContinuousControl": ["Pump", "Outlet"],
+    "PidControl": ["Pump", "Outlet"],
+    "UserDemand": ["Basin", "Terminal", "LevelBoundary"],
+    "LevelDemand": ["Basin"],
+    "FlowDemand": [
+        "LinearResistance",
+        "ManningResistance",
+        "TabulatedRatingCurve",
+        "Pump",
+        "Outlet",
+    ],
+}
+
+
+# Function to validate connection
+def can_connect(node_up: str, node_down: str) -> bool:
```

Comment:
Maybe `can_node_type_connect`? And for the arguments, make clear it's about the type, e.g. `node_type_up`?

Reply:
I think after changing the arguement names, it is very clear that this function is returning if two node type can be connected. I also update the comment. So maybe the function name can stay `can_connect`?